WooCommerce/Classes/Authentication/Epilogue/StorePickerCoordinator.swift

```diff
@@ -35,16 +40,35 @@ final class StorePickerCoordinator: Coordinator {
         return pickerVC
     }()
 
-    init(_ navigationController: UINavigationController, config: StorePickerConfiguration) {
+    init(_ navigationController: UINavigationController,
+         config: StorePickerConfiguration,
+         switchStoreUseCase: SwitchStoreUseCaseProtocol = SwitchStoreUseCase(stores: ServiceLocator.stores)) {
         self.navigationController = navigationController
         self.selectedConfiguration = config
+        self.switchStoreUseCase = switchStoreUseCase
     }
 
     func start() {
         showStorePicker()
     }
-}
 
+    /// Checks available stores and switch store if only one valid store is available
+    ///
+    @MainActor
+    func switchStoreIfOnlyOneStoreIsAvailable() async throws {
+        let sites = switchStoreUseCase.getAvailableStores()
+
+        guard let siteID = sites.first?.siteID else {
+            throw StoreSelectionError.noStoresAvailable
+        }
+
+        guard !sites.containsMoreThanOne else {
+            throw StoreSelectionError.moreThanOneStoreAvailable
```

Comment:
nit: it feels a bit surprising to me for these scenarios to be an "error," and `moreThanOneStoreAvailable` error isn't handled explicitly. how about renaming `StoreSelectionError` to `OneStoreSwitchError` (something like this) or just returning a boolean?